### DISCUSSION TITLE
Updating relational field

### DISCUSSION THREAD

**Discussion by andriivasylchenko** [Budibase/budibase#3379] [Help]
Hello there! I've just started exploring budibase and trying to figure out how to update relational field dropdown. 

Some details:
I'm building a page with 2 forms (form1, form2). Each form is creating an entry in budibase database tables (table1 and table2 respectively). The challenge is that table1 has a relational field to table1 (and vise versa). That means when I'm adding an entry through form1, form2 should somehow update itself, otherwise, that relational dropdown field won't show new entry added by form1.

I found a few workarounds with separating forms with different pages or opening them in modals, but ideally, I would like to find a way to refresh form2. I've tried playing with including one provider into another, updating state and etc, but seems like this has no effect because form itself is set to a table, not a provider (if to set form to a provider, dropdown becomes empty).

What should be a proper way of solving this issue?

Thanks in advance for any reply!

**Comment by mjashanks** [comment]
Hi There. Just to clarify, you want to
1. Create a new record for table 1
2. Then, immediately create a new record for table 2, which is related to the record created in step 1

At the moment, the only way to do this would be as 2 separate screens.
1. Open create screen for `table 1` 
2. On the "Save" action for this screen, you could navigate directly to the create screen for table 2 - `/table2/new/row`
3. The relationship picker should now contain the new `table 1` record

However, [this issue](https://github.com/Budibase/budibase/issues/3383) has just been logged, which would allow you to do everything in one screen and use the `_id` of the first record directly, when saving to `table 2` - meaning that the relationship picker is not required.
